README.md

```diff
@@ -0,0 +1,434 @@
+[![](https://travis-ci.org/buildo/avenger.svg)](https://travis-ci.org/buildo/avenger)
+[![](https://img.shields.io/npm/v/avenger.svg?sytle=flat-square)](https://www.npmjs.com/package/avenger)
+[![npm downloads](https://img.shields.io/npm/dm/avenger.svg?style=flat-square)](https://www.npmjs.com/package/avenger)
+[![](https://david-dm.org/buildo/avenger.svg)](https://david-dm.org/buildo/avenger#info=dependencies&view=list)
+[![](https://david-dm.org/buildo/avenger/dev-status.svg)](https://david-dm.org/buildo/avenger#info=devDependencies&view=list)
+
+---
+
+# Intro
+
+Avenger is a data fetching and caching layer written in TypeScript, it aims at implementing the principle of **Command Query Responsibility Segregation** without necessarily enforcing it.
+If you are new to the concept you can get a grasp of its foundations in [this nice aricle](https://martinfowler.com/bliki/CQRS.html) by Martin Fowler. Using his words:
+
+_"The change that CQRS introduces is to separate models for update and display, which it refers to as "Command" and "Query" respectively following the vocabulary of CommandQuerySeparation._
+
+_The rationale is that for many problems, particularly in more complicated domains, having the same conceptual model for commands and queries leads to a more complex model that does neither well."_
+
+# Avenger
+At the heart of the DSL of the software there are two constructors: **query** and **command**.
+
+## queries
+The [**`query`**](#query) function allows you to query your data source and get an object of type [**`CachedQuery`**](#CachedQuery) in return.
+It accepts two parameters, the first is a function with a [**`Fetch`**](#Fetch) signature that is used to retrive data from your data source, the second is an object with the[**`Strategy`**](#Strategy) signature that will be used to decide if the data stored by **Avenger** is still relevant or needs to be refetched.
+
+Although important, `query` is a pretty low level API and **Avenger** offers some convenient utils with a [**`StrategyBuilder`**](#StrategyBuilder) signature that you should prefer over it (unless you have very specific needs):
+
+- **refetch:** runs the fetch function everytime the data is requested.
+- **expire:** when the data is requested, the fetch function is run only if data in the `Cache` is older than the expiration defined, otherwise the cached value is used.
+- **available:** when the data is requested, if a cahed value is available it is always returned, otherwise the fetch function is run and the result stored in the `Cache` accordingly.
+
+All these utils ask you to specify custom `Setoid` instances for the inputs and outputs of the `Fetch` function but you can (and should) use them toghether with one of the built in implementations:
```

Comment:
Not clear the `why` and it requires to know what a `Setoid` is.
"will use *** equality check" -> to check what?

Reply:
I put a link to `fp-ts` docs for Setoid and tried to be a little clearer on what they are used for